### PR TITLE
feat: update checker.

### PR DESCRIPTION
**Pull request by rmdwe**

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,6 +1,16 @@
 /** @type {import('next').NextConfig} */
+const { version } = require('./package.json');
+
 const nextConfig = {
   reactStrictMode: true,
+  webpack: (config, { webpack }) => {
+    config.plugins.push(
+      new webpack.DefinePlugin({
+        'process.env.CONFIG_CURRENT_VERSION': JSON.stringify(version),
```
Comment (Cyronlee) [inline next.config.js:9]: could be a better name?

File: src/components/UpdateChecker.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  const currentVersion = process.env.CONFIG_CURRENT_VERSION || '0.0.0';
+  const infoToast = useInfoToast();
+
+  useEffect(() => {
+    fetch('https://raw.githubusercontent.com/Cyronlee/zBoard/main/package.json')
+      .then((res) => res.json())
+      .then((data) => {
+        const latestVersion = data.version;
+        if (compareVersions(latestVersion, currentVersion)) {
+          infoToast('New update available', 'Please update.');
```
Comment (Cyronlee) [inline src/components/UpdateChecker.tsx:14]: can we show the current version & newest version in the toast, and add a link to https://github.com/Cyronlee/zBoard/releases

File: src/components/UpdateChecker.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        }
+      })
+      .catch(() => console.warn('Fetch latest version failed.'));
+  }, []);
+  return <></>;
+};
+
+const compareVersions = (v1: string, v2: string) => {
```
Comment (Cyronlee) [inline src/components/UpdateChecker.tsx:22]: can we just compare string like '1.13.0' > '1.9.0'

File: src/lib/customToast.ts [hunk 4]
```diff
@@ -14,3 +14,17 @@ export const useErrorToast = () => {
     });
   };
 };
+
+export const useInfoToast = () => {
```
Comment (Cyronlee) [inline src/lib/customToast.ts:18]: Could define a special toast, with a long auto close time

File: src/components/UpdateChecker.tsx (hunk 3 above)
Comment (rmdwe) [reply src/components/UpdateChecker.tsx:22]: > '1.1'>'1.90'
> -> false

File: src/lib/customToast.ts (hunk 4 above)
Comment (rmdwe) [reply src/lib/customToast.ts:18]: '1.10'>'1.102'
-> false

**Review by Cyronlee — APPROVED** [review]

**Pull request merged**